### PR TITLE
Host/target corrections and clarifications and cleanup.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: mono/mini/local-propagation.c [hunk 1]
```diff
@@ -677,7 +677,7 @@ mono_local_cprop (MonoCompile *cfg)
 						mono_inst_set_src_registers (ins, sregs);
 
 						if ((opcode2 == OP_VOIDCALL) || (opcode2 == OP_CALL) || (opcode2 == OP_LCALL) || (opcode2 == OP_FCALL))
-							((MonoCallInst*)ins)->fptr = (gpointer)ins->inst_imm;
+							((MonoCallInst*)ins)->fptr = (gpointer)(uintptr_t)ins->inst_imm;
```
Comment (jaykrell) [inline mono/mini/local-propagation.c]: `intptr_t` or `uintptr_t`?

File: mono/mini/ssa.c [hunk 2]
```diff
@@ -1116,7 +1116,7 @@ fold_ins (MonoCompile *cfg, MonoBasicBlock *bb, MonoInst *ins, MonoInst **carray
 				ins->sreg2 = -1;
 
 				if ((opcode2 == OP_VOIDCALL) || (opcode2 == OP_CALL) || (opcode2 == OP_LCALL) || (opcode2 == OP_FCALL))
-					((MonoCallInst*)ins)->fptr = (gpointer)ins->inst_imm;
+					((MonoCallInst*)ins)->fptr = (gpointer)(uintptr_t)ins->inst_imm;
```
Comment (jaykrell) [inline mono/mini/ssa.c]: `intptr_t` or `uintptr_t`?

File: mono/mini/exceptions-x86.c [hunk 3]
```diff
@@ -858,7 +858,7 @@ mono_arch_unwind_frame (MonoDomain *domain, MonoJitTlsData *jit_tls,
 	} else if (*lmf) {
 		g_assert ((((gsize)(*lmf)->previous_lmf) & 2) == 0);
 
-		if ((ji = mini_jit_info_table_find (domain, (gpointer)(*lmf)->eip, NULL))) {
+		if ((ji = mini_jit_info_table_find (domain, (gpointer)(uintptr_t)(*lmf)->eip, NULL))) {
```
Comment (jaykrell) [inline mono/mini/exceptions-x86.c]: `intrptr_t` or `uintptr_t`?

File: mono/mini/mini-generic-sharing.c [hunk 4]
```diff
@@ -1945,13 +1945,13 @@ mini_get_gsharedvt_wrapper (gboolean gsharedvt_in, gpointer addr, MonoMethodSign
 	if (mono_aot_only)
 		addr = mono_aot_get_gsharedvt_arg_trampoline (info, addr);
 	else
-		addr = mono_arch_get_gsharedvt_arg_trampoline (mono_domain_get (), info, addr);
+		addr = mono_arch_get_gsharedvt_arg_trampoline (mono_domain_get (), (target_mgreg_t)(uintptr_t)info, (target_mgreg_t)(uintptr_t)addr);
```
Comment (jaykrell) [inline mono/mini/mini-generic-sharing.c]: `intptr_t` or `uintptr_t`?

**Comment by jaykrell** [comment]
@monojenkins build Windows i386
@monojenkins build Linux x64
@monojenkins build Linux x64 C++
@monojenkins build OS X i386

**Comment by jaykrell** [comment]
@monojenkins build Linux x64

**Comment by jaykrell** [comment]
@monojenkins build Linux x64 C++

**Comment by jaykrell** [comment]
@monojenkins build OS X i386

**Comment by jaykrell** [comment]
@monojenkins build Windows i386

File: mono/mini/arch-stubs.c [hunk 5]
```diff
@@ -22,7 +22,7 @@ mono_arch_get_gsharedvt_call_info (gpointer addr, MonoMethodSignature *normal_si
 }
 
 gpointer
-mono_arch_get_gsharedvt_arg_trampoline (MonoDomain *domain, gpointer arg, gpointer addr)
+mono_arch_get_gsharedvt_arg_trampoline (MonoDomain *domain, target_mgreg_t arg, target_mgreg_t addr)
 {
```
[the remark this answers is not in the document]
Comment (jaykrell) [reply mono/mini/arch-stubs.c]: Ok, I wasn't sure. In the past you had me do things that generate code, look like JIT. I'll revert that.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build Windows i386

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by luhenry** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build Windows i386

File: mono/tests/libtest.c [hunk 6]
```diff
@@ -5630,8 +5630,7 @@ mono_test_marshal_return_single_double_struct (void)
 	return res;
 }
 
-
-#ifndef TARGET_X86
+#ifndef HOST_X86
```
Comment (jaykrell) [inline mono/tests/libtest.c]: Need to undo this line, in this PR, and deal with it separately. This single line change breaks tests.

**Comment by jaykrell** [comment]
System mono  crashes on arm64.  Bug is fixed but not in system  mono.

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**